webserver.py

```diff
@@ -0,0 +1,323 @@
+			self.shutdown = False
+
+			
+
+
+class MusicControl(object):
+	def __init__(self):
+		self.musicChild = None
+
+
+	def startEffect(self, effect):
+		self.musicChild = subprocess.Popen([os.path.dirname(os.path.abspath(__file__))+'/main.py', '--effect', effect], stdin=subprocess.PIPE)
+		print 'Spawning hyperion-audio-effect --effect '+effect
+
+
+	def kill(self):
+		if self.musicChild is None:
+			print "Music Child does not exist."
+			return
+
+		self.musicChild.communicate(input='x\n')
+		self.musicChild.wait()
+		self.musicChild = None
+		print "Music Child terminated."
+
+
+
+class LightControl(object):
+	def __init__(self):
+		self.active = True
+		self.stopThread = False
+		self.interval = 3
+		self.countDown = self.interval
+		self.checkLEDStatus()
+
+	def checkLEDStatus(self):
+		if self.stopThread:
+			return
+
+		threading.Timer(1, self.checkLEDStatus).start()
+		if self.active:
+			self.countDown -= 1
+			if self.countDown == 0:
+				self.countDown = self.interval
+				process = subprocess.Popen(['hyperion-remote', '-c', 'ffd385'])
+		else:
+			self.countDown = self.interval
+
+
+	def stop(self):
+		self.stopThread = True
+		
+
+
+
+#-- The class is reinstantiated every time, a request is received
+class HTTPRequest(BaseHTTPRequestHandler):
+
+	def __init__(self, request, client_address, server):
+		self.menu = {'Main Menu': '/', 'Video Effects': 'getEffectList.html', 'Music Effects': 'musicEffects.html'}
+		BaseHTTPRequestHandler.__init__(self, request, client_address, server)
+
+	#Handler for the GET requests
+	def do_GET(self):
+
+		# Send the html message
+		if self.path == '/':
+			self.getIndexFile()
+		elif self.path.startswith('/style.css'):
+			self.getCSSstyle()
+		elif self.path.startswith('/getEffectList.html'):
+			self.getEffectList()
+		elif self.path.startswith('/musicEffects.html'):
+			self.musicEffects()
+		elif self.path.startswith('/shutdown.html'):
+			self.shutDown()
+		elif self.path.startswith('/favicon.ico'):
+			self.favicon()
+		else:
+			print self.path+" not found."
+			self.send_error(404, "File ("+self.path+") not found")
+
+
+	def getIndexFile(self):
+		self.sendHeader()
+		self.wfile.write("<h1>Main Menu</h1>")
+		self.wfile.write("<nav><ul>")
+
+		
+		keys = self.menu.keys()
+		for i in keys:
+			self.wfile.write("<li><form action=\""+self.menu[i]+"\"><input type=\"submit\" value=\""+i+"\"></form></li>")
+
+		self.wfile.write("<li><form action=\"shutdown.html\"><input type=\"submit\" value=\"Shutdown\"></form></li>")
+		self.wfile.write("</nav></ul>")
+		self.sendFooter();
+
+
+	def getCSSstyle(self):
+		self.send_response(200)
+		self.send_header('Content-type', 'text/css')
+		self.end_headers()
+		self.getFile('style.css')
+		self.sendFooter();
+
+
+
+	def musicEffects(self):
+		effects = []
+		for effect in os.listdir(os.path.dirname(os.path.abspath(__file__))+"/effects"):
+			if effect.endswith(".json"):
+				effects.append(effect[:-5])
```

Comment:
Would be better to read in the `json` and use `script` field for commands and `name` field for UI.